### ISSUE TITLE
Needs to be a way to clear data

### ISSUE THREAD

**Issue by daveconger** [swri-robotics/mapviz#574]
My mapviz keep getting filled up with clutter from successive bag runs.  There needs to be a way to clear data (or re-init plugins).  Restarting mapviz each time is not sufficient since geo maps are not cached and need to redownload each run.

**Comment by HoomanBean** [comment]
Yeah the tubes keep getting clogged. Also I need to borrow Jerry's scissors.

**Comment by malban** [comment]
Seems like a useful feature.  Base plugin could be modified with a virtual clear data method and individual plugins could be modified to clear their buffers.

What specifically is getting cluttered up?  Things like the odometry track?

**Comment by malban** [comment]
Also  @HoomanBean  doesn't seem like a real person, I wouldn't trust anything they say.

**Comment by daveconger** [comment]
Off hand, at least the gps display.

**Comment by malban** [comment]
@daveconger See if these branches work:

https://github.com/malban/mapviz/tree/clear-history-indigo
https://github.com/malban/mapviz/tree/clear-history-jade

It adds a 'Clear History' button to the menu bar and clears the history of GPS, tf, and odometry tracks, point clouds, laser scans, and markers.  You will need to recompile both mapviz and mapviz_plugins.

**Comment by HoomanBean** [comment]
cool beans

**Comment by danthony06** [comment]
Closing as stale.
